app/invocation/invocation_model.tsx

```diff
@@ -264,42 +268,46 @@ export default class InvocationModel {
 
     if (!groups?.length) return null;
 
-    const invocation = this.invocations[0];
-    if (!invocation) return null;
-
+    const invocation = this.getPrimaryInvocation();
     return groups.find((group) => group.id === invocation.acl?.groupId) || null;
   }
 
   isAnonymousInvocation(): boolean {
-    return this.invocations.find(() => true)?.acl?.groupId === "";
+    return this.getPrimaryInvocation().acl?.groupId === "";
   }
 
   hasCacheWriteCapability(): boolean {
     return Boolean(
-      this.invocations
-        .find(() => true)
-        ?.createdWithCapabilities?.some(
-          (existingCapability) =>
-            existingCapability == api_key.ApiKey.Capability.CACHE_WRITE_CAPABILITY ||
-            existingCapability == api_key.ApiKey.Capability.CAS_WRITE_CAPABILITY
-        )
+      this.getPrimaryInvocation().createdWithCapabilities?.some(
+        (existingCapability) =>
+          existingCapability == api_key.ApiKey.Capability.CACHE_WRITE_CAPABILITY ||
+          existingCapability == api_key.ApiKey.Capability.CAS_WRITE_CAPABILITY
+      )
     );
   }
 
-  getId(): string | undefined {
-    return this.invocations.find(() => true)?.invocationId;
+  getId(): string {
+    return this.getPrimaryInvocation().invocationId;
+  }
+
+  getPrimaryInvocation(): invocation.Invocation {
+    return this.invocations[0];
+  }
+
+  getInvocations(): invocation.Invocation[] {
+    return this.invocations;
   }
```

Comment:
Instead of `getPrimaryInvocation()`, I think we could have the model expose a public readonly `invocation` instead of keeping a list of invocations. I don't think we only ever look at `invocations[0]` currently, and if we ever want more than one, then it'll make more sense to create one model per invocation rather than having a list